upgrades/upgrade_testing_framework/steps/step_restore_source_snapshot.py

```diff
@@ -1,58 +1,48 @@
 import json
 import time
 
+import upgrade_testing_framework.clients as clients
 from upgrade_testing_framework.core.framework_step import FrameworkStep
-import upgrade_testing_framework.core.shell_interactions as shell
 
 class RestoreSourceSnapshot(FrameworkStep):
     """
     This step restores the source snapshot on the target cluster and confirms it worked.
     """
 
     def _run(self):
-        """
-        The contents of this are quick/dirty.  Instead of raw curl commands we should probably be using the Python SDK
-        for Elasticsearch/OpenSearch.  At the very least, we should wrap the curl commands in our own pseudo-client.
-        """
-
         # Get the state we need
         shared_volume = self.state.shared_volume
+        snapshot = self.state.snapshot
+        source_doc_id = self.state.source_doc_id
         target_cluster = self.state.target_cluster
 
         # Begin the step body
         self.logger.info("Checking if source snapshots are visible on target...")
-        port = target_cluster.rest_ports[0]
-
-        register_cmd = f"""curl -X PUT 'localhost:{port}/_snapshot/noldor' -H 'Content-Type: application/json' -d '{{
-            "type": "fs",
-                "settings": {{
-                    "location": "{shared_volume.mount_point}"
-            }}
-        }}'
-        """
-        _, _ = shell.call_shell_command(register_cmd)
-
-        check_cmd = f"curl -X GET 'localhost:{port}/_snapshot/noldor/_all?pretty' -ku 'admin:admin'"
-        _, output = shell.call_shell_command(check_cmd)
-        if "noldor" in "\n".join(output):
+        node = target_cluster.nodes[0] # shouldn't matter which node we pick
+        port = node.rest_port
+        engine_version = node.engine_version
+        rest_client = clients.get_rest_client(engine_version)
+
+        rest_client.register_snapshot_dir(port, snapshot.repo_name, shared_volume.mount_point)
+        response_all_snapshots = rest_client.get_snapshots_all(port, snapshot.repo_name)
+
+        if "noldor" in response_all_snapshots.response_text:
             self.logger.info("Source snapshot visible to target cluster")
         else:
             self.fail("Snapshot restoration failed; source snapshots not visible to target cluster")
 
         self.logger.info("Restoring source snapshot onto target...")
-        restore_cmd = f"curl -X POST 'localhost:{port}/_snapshot/noldor/1/_restore?pretty' -ku 'admin:admin'"
-        _, _ = shell.call_shell_command(restore_cmd)
+        rest_client.restore_snapshot(port, snapshot.repo_name, snapshot.snapshot_id)
 
         self.logger.info("Waiting a few seconds for the snapshot to be restored...")
-        time.sleep(5)
+        time.sleep(3)
```

Comment:
I imagine in the future we want to replace this with a REST call that is checking for the proper status

Reply:
Agreed, this is a hack.  For stuff like this my philosophy is to fix it when we have the inclination or the need, and repeat that process in successive waves whose direction is pointed at a specific demo deliverable.

The code I'm most inclined to be permissive of issues in is the FrameworkStep code (barring some exceptions, like how the global state is interacted with).  That's because the I'm confident we'll find a use for the library code surrounding the FrameworkSteps, but the specific steps we're working on may change rapidly based on feedback.  As a result, I'm not even bothering to unit test the FrameworkStep implementations until we have confidence in the direction things are heading in and have buy-in w/ management and users.  I'm unit testing the library code heavily because I do have confidence in its long-term value.

Reply:
@mikaylathompson FYI ^^